src/test/testStore.tsx

```diff
@@ -39,3 +39,22 @@ export function renderWithProviders(
     user: userEvent.setup(),
   };
 }
+
+export function renderWithStoreProvider(
+  ui: React.ReactElement,
+  {
+    preloadedState = {},
+    // Automatically create a store instance if no store was passed in
+    store = setupStore(preloadedState),
+    ...renderOptions
+  }: ExtendedRenderOptions = {},
+) {
+  function Wrapper({ children }: PropsWithChildren): JSX.Element {
+    return <Provider store={store}>{children}</Provider>;
+  }
+  return {
+    store,
+    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
+    user: userEvent.setup(),
+  };
+}
```

Comment:
Why create a whole new provider? You could just extend the current provider.
It is totally fine IMO :)

Reply:
In other glance, what does this do? We render it without BrowserRouter and use it in Navigation-test? Does the test then actually test the navigation? I think not

Reply:
AHa, now I got it. The Navigation-already contains BrowserRouter. Then OK. I think it is better.

But in this case, why not just remove the BrowserRouter from the existing provider-function? And then in those tests that have BrowserRouter, remove it. Everything should work

Reply:
Having two provider functions will start a momentum for the unit-test -setup to go to very confusing direction.

Reply:
So for example, the MentorPages test could look like this:

```
  it('displays fetched mentors in list and can open the card for more information', async () => {
    const {
      user,
      getByRole,
      queryByRole,
      findByRole,
      getByText,
      getAllByRole,
    } = renderWithProviders(
      <>
        <Navigation />
        <AppToaster />
      </>,
    );
    const link = getByRole('link', { name: 'navigation.mentors' });
    user.click(link);

// continue test below normally
```

Reply:
Also sidenote. I run the unit tests not so long ago, but I dont remember there being hundreds of warnings of deprecation? Yikes :E

Reply:
It has started happening in this PR, since the https://github.com/sos-lapsikyla/ylitse-web/actions/runs/11495612541/job/31995497788 run still has no warnings.

Reply:
Flooding this same discussion, some of the test warnings are easier to mitigate than others. I just gave it a try and it seems a bit funked.

But for example in Navigation.test it is simple as 

```typescript
// wrap act around this change
    await act(async () => {
      store.dispatch(mentorsApi.endpoints.getMentors.initiate());
      jest.advanceTimersByTime(1000);
    });
```

There are more problematic warnings to fix, for example pages that are wrapped around <PageWithTransition> component
One option for these could be to Move the PageWithTransition around single-components, and instead use it around as a general Higher Order Component / Provider as like the ConfirmationDialogProvider
Its also a bit silly that we wrap individual pages around with the PageWithTransition, when we could have one place for it? And then we can drop if from the `renderWithProviders` function

The punycode one, I did not look into it yet.

Reply:
It seems like the warnings appeared when I had to updated some npm packages. We are still using many versions not up to date.

Reply:
That refactor is a good idea! But it has to be dealt with later, because now my time is running out!